File: r.php
<?php
$config['mysql_host'] = 'localhost';
$config['mysql_user'] = 'asteriskuser';
$config['mysql_password'] = 'amp109';
$config['mysql_base'] = 'telnumbers';
$mysql_table = 'codes';

if(isset($_GET['number'])) {
        $number=$_GET['number'];
} else {
        die("Номера нема");
}

$mysql = new mysqli($config['mysql_host'],$config['mysql_user'],$config['mysql_password'],$config['mysql_base']);
$mysql->query("SET NAMES 'utf8'");

//$number = '+7(952)44-044-17';

$number = preg_replace('/[^0-9]+/','',$number);

if (strlen($number) == 11) {
  $number = substr($number,1);
}

if (strlen($number) == 10) {
  $def = substr($number,0,3);
  $code = substr($number,3);
  $stmt = $mysql->stmt_init();
  $stmt->prepare(
  //'SELECT `operator`,`region` FROM `codes` WHERE 4404417 BETWEEN `code_from` AND `code_to` AND `code_abcdef`=952;'
  //'SELECT `region`,`operator` FROM `codes` WHERE `code_abcdef`= ? AND ? > `code_from` AND ? < `code_to`'
  'SELECT `region` FROM `codes` WHERE `code_abcdef`= ? AND ? > `code_from` AND ? < `code_to`'
   );
  //$stmt->bind_param("iii", $def, $code, $code);
  $stmt->bind_param("iii", $def, $code, $code);
  $stmt->execute();
  //$stmt->bind_result($region, $operator);
  $stmt->bind_result($region);
  while ($stmt->fetch()) {
	  if ($region === "Нижегородская область"){
		  echo "$number<br/>\r\n";
	  }else {
		  echo "@$number<br/>\r\n";
	  }
  }
  $stmt->close();
} else {
  echo "Ошибка: неправильный формат номера<br>\r\n";
}
?>

File: r.pl
#!/usr/bin/env perl
#[ivrsql]
#exten => *777,1,Answer()
#exten => *777,n,Wait(1)
#exten => *777,n,AGI(sqlmenu.agi)
#exten => *777,n,Hangup()
use warnings;
use strict;
use Asterisk::AGI;
use DBI;
#$|=1;

my $AGI = new Asterisk::AGI;
my %input = $AGI->ReadParse();

### Play "welcome.gsm", greeting the caller 

$AGI->stream_file('welcome');

### Play "product-code-prompt.gsm", instructing the caller to enter the 
### product code, then allow ten seconds to enter four-digits

my $prodcode = $AGI->get_data('product',10000,4);
my $quantity = &quan_by_code($prodcode);

#$AGI->verbose("prodcode=$prodcode\n", 1);


sub quan_by_code {
### Takes a product code as input, then returns the quantity available.
        my $code = shift;
        my $dbh = open_connection();
        #//'SELECT `operator`,`region` FROM `codes` WHERE 4404417 BETWEEN `code_from` AND `code_to` AND `code_abcdef`=952;'
		#//'SELECT `region`,`operator` FROM `codes` WHERE `code_abcdef`= ? AND ? > `code_from` AND ? < `code_to`'
        'SELECT `region` FROM `codes` WHERE `code_abcdef`= ? AND ? > `code_from` AND ? < `code_to` LIMIT 1'
        my $sql = "SELECT code, quantity FROM product WHERE code=\'$code\' LIMIT 1";
        my $sth = $dbh->prepare($sql);
        $sth->execute or die "Unable to execute SQL query: $dbh->errstr\n";
        my $row = $sth->fetchrow_arrayref;
        $sth->finish;
        $dbh->disconnect;
        if ( ($row->[0] == "Нижегородская область"){
                return 1;
        } else {
                return 0;
        }
}

sub open_connection {
        my $database = "telnumbers";
        my $hostname = "localhost";
        my $username = "asteriskuser";
        my $password = "amp109";
        my $dsn = "mysql:$database:$hostname";
        return DBI->connect("DBI:$dsn",$username,$password) or die $DBI::errstr;
}
